File: tests/integration/aliases/go/rename/step1/main.go
// Copyright 2016-2020, KhulnaSoft Ltd.  All rights reserved.
//go:build !all
// +build !all

package main

import (
	"github.com/khulnasoft/codeinfra/sdk/v3/go/codeinfra"
)

// FooComponent is a component resource
type FooComponent struct {
	codeinfra.ResourceState
}

func main() {
	codeinfra.Run(func(ctx *codeinfra.Context) error {
		fooComponent := &FooComponent{}
		return ctx.RegisterComponentResource("foo:component", "foo", fooComponent)
	})
}

File: tests/integration/aliases/python/rename/step1/__main__.py
# Copyright 2016-2018, KhulnaSoft Ltd.  All rights reserved.

from codeinfra import Alias, ComponentResource, export, Resource, ResourceOptions, create_urn, ROOT_STACK_RESOURCE

class Resource1(ComponentResource):
    def __init__(self, name, opts=None):
        super().__init__("my:module:Resource", name, None, opts)

# Scenario #1 - rename a resource
res1 = Resource1("res1")
